test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+use futures::stream::Stream;
+use futures::StreamExt;
+// use opentelemetry::global::shutdown_tracer_provider;
+use opentelemetry::sdk::metrics::{selectors, PushController};
+// use opentelemetry::trace::TraceError;
+use opentelemetry::global;
+use opentelemetry::metrics::{self, ObserverResult, ValueRecorder};
+use opentelemetry_otlp::ExporterConfig;
+use std::error::Error;
+use std::time::Duration;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
```

Comment:
I got this to build without any locks. We should talk through it tomorrow I think.

Reply:
As we discussed offline (documenting here in case others are curious), it compiles when you remove the Mutexes but the behavior is not the same. Since the `f64` type implements the `Copy` trait, the `move` on the observer instrument's callback closure results in a copy of the variable that holds the value of the metric. When that metric variable is updated in the main thread, the callback closure does not see these updates reflected because it's using a copy of the variable.

I looked into using Atomics but unfortunately there is no `AtomicF64` in the `std::sync::atomics` crate. There is a non-std crate that has an AtomicF64 that I'm reluctant to pull in as a dependency, but we can consider it in the future if we find that the locks result in too much of a performance hit.